webapp/src/js/bootstrapper/initial-replication.js

```diff
@@ -0,0 +1,188 @@
+const utils = require('./utils');
+const translator = require('./translator');
+const { setUiStatus } = require('./ui-status');
+
+let docIdsRevs;
+let lastSeq;
+let remoteDocCount;
+
+const INITIAL_REPLICATION_LOG = '_local/initial-replication';
+const BATCH_SIZE = 100;
+
+const startInitialReplication = async (localDb) => {
+  if (await getReplicationLog(localDb)) {
+    return;
+  }
+
+  const log = {
+    _id: INITIAL_REPLICATION_LOG,
+    start_time: Date.now(),
+    start_data_usage: getDataUsage(),
+  };
+
+  await localDb.put(log);
+};
+
+const completeInitialReplication = async (localDb) => {
+  const dbSyncEndData = getDataUsage();
+
+  const replicationLog = await getReplicationLog(localDb);
+  if (!replicationLog) {
+    throw new Error('Invalid replication state: missing replication log');
+  }
+
+  replicationLog.complete = true;
+  replicationLog.duration =  Date.now() - replicationLog.start_time;
+  replicationLog.data_usage = replicationLog.start_data_usage &&
+                              dbSyncEndData.app.rx - replicationLog.start_data_usage.app.rx;
+
+  console.info('Initial sync completed successfully in ' + (replicationLog.duration / 1000) + ' seconds');
+  if (replicationLog.data_usage) {
+    console.info('Initial sync received ' + replicationLog.data_usage + 'B of data');
+  }
+
+  await localDb.put(replicationLog);
+};
+
+const getDataUsage = () => {
+  if (window.medicmobile_android && typeof window.medicmobile_android.getDataUsage === 'function') {
+    return JSON.parse(window.medicmobile_android.getDataUsage());
+  }
+};
+
+const displayTooManyDocsWarning = ({ warn_docs, limit }) => {
```

Comment:
good fit for ui-status.js?

Reply:
Yea, I agree! I'll move it! Thanks for noticing.